packages/react/src/components/Footer/LocaleButton.js

```diff
@@ -0,0 +1,70 @@
+import { FOOTER_LOCALE_BUTTON } from '../../internal/FeatureFlags.js';
+import { featureFlag } from '@carbon/ibmdotcom-utilities';
+
+import React, { useState } from 'react';
+import {
+  Button,
+  ComposedModal,
+  ModalHeader,
+  ModalBody,
+} from 'carbon-components-react';
+import { settings } from 'carbon-components';
+import { Globe20 } from '@carbon/icons-react';
+
+const { prefix } = settings;
+
+/**
+ * EXPERIMENTAL: Renders the locale button
+ *
+ * @private
+ * @returns {object} JSX object
+ */
+const LocaleButton = () => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  return featureFlag(
+    FOOTER_LOCALE_BUTTON,
+    <div className={`${prefix}--locale-btn__container`}>
+      <Button
+        data-autoid="locale-btn"
+        className={`${prefix}--locale-btn`}
+        kind="secondary"
+        onClick={open}
+        renderIcon={Globe20}>
+        United States — English
+      </Button>
+
+      <ComposedModal open={isOpen} onClose={close} data-autoid="locale-modal">
+        <ModalHeader
+          label="United States — English"
+          title="Select your region"
+        />
+        <ModalBody>
+          <p className={`${prefix}--modal-content__text`}>
+            Placeholder text for now
+          </p>
+        </ModalBody>
+      </ComposedModal>
+    </div>
+  );
+
+  /**
+   * Sets modal state to open
+   *
+   * @private
+   */
+  function open() {
```

Comment:
@photodow Should we standardize placement of functions like this purely for consistency? Only because all [Masthead](https://github.com/carbon-design-system/ibm-dotcom-library/blob/master/packages/react/src/components/Masthead/MastheadSearch.js) functions like this come at the start (and mine are longer haha)

Reply:
@kennylam this was merged in already, I think we can continue the conversation on this in https://github.com/carbon-design-system/ibm-dotcom-library/pull/75 and make any changes if necessary.